samples/core/Querying/UserDefinedFunctionMapping/Model.cs

```diff
@@ -0,0 +1,166 @@
+        public double PercentageDifference(int first, int second)
+            => throw new NotSupportedException();
+        #endregion
+
+        #region QueryableFunctionDefinition
+        public IQueryable<Post> PostsWithPopularComments(int likeThreshold)
+            => FromExpression(() => PostsWithPopularComments(likeThreshold));
+        #endregion
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            #region EntityConfiguration
+            modelBuilder.Entity<Blog>()
+                .HasMany(b => b.Posts)
+                .WithOne(p => p.Blog);
+
+            modelBuilder.Entity<Post>()
+                .HasMany(p => p.Comments)
+                .WithOne(c => c.Post);
+            #endregion
+
+            modelBuilder.Entity<Blog>()
+                .HasData(
+                    new Blog { BlogId = 1, Url = @"https://devblogs.microsoft.com/dotnet", Rating = 5 },
+                    new Blog { BlogId = 2, Url = @"https://mytravelblog.com/", Rating = 4 });
+
+            modelBuilder.Entity<Post>()
+                .HasData(
+                    new Post { PostId = 1, BlogId = 1, Title = "What's new", Content = "Lorem ipsum dolor sit amet", Rating = 5 },
+                    new Post { PostId = 2, BlogId = 2, Title = "Around the World in Eighty Days", Content = "consectetur adipiscing elit", Rating = 5 },
+                    new Post { PostId = 3, BlogId = 2, Title = "Glamping *is* the way", Content = "sed do eiusmod tempor incididunt", Rating = 4 },
+                    new Post { PostId = 4, BlogId = 2, Title = "Travel in the time of pandemic", Content = "ut labore et dolore magna aliqua", Rating = 3 });
+
+            modelBuilder.Entity<Comment>()
+                .HasData(
+                    new Comment { CommentId = 1, PostId = 1, Text = "Exciting!", Likes = 3 },
+                    new Comment { CommentId = 2, PostId = 1, Text = "Dotnet is useless - why use C# when you can write super fast assembly code instead?", Likes = 0 },
+                    new Comment { CommentId = 3, PostId = 2, Text = "Didn't think you would make it!", Likes = 3 },
+                    new Comment { CommentId = 4, PostId = 2, Text = "Are you going to try 70 days next time?", Likes = 5 },
+                    new Comment { CommentId = 5, PostId = 2, Text = "Good thing the earth is round :)", Likes = 5 },
+                    new Comment { CommentId = 6, PostId = 3, Text = "I couldn't agree with you more", Likes = 2 });
+
+            #region BasicFunctionConfiguration
+            modelBuilder.HasDbFunction(typeof(BloggingContext).GetMethod(nameof(ActivePostCountForBlog), new[] { typeof(int) }))
+                .HasName("CommentedPostCountForBlog");
+            #endregion
+
+            #region HasTranslationFunctionConfiguration
+            // 100 * ABS(first - second) / ((first + second) / 2)
+            var doubleTypeMapping = new SqlServerDoubleTypeMapping("float");
```

Comment:
You can avoid type mapping and associated warning if you make first/second double in C# and just reuse their type mapping.

Reply:
don't we want to show people how to create the mapping themselves? like when you divide/multiply by a double constant, you would have to do it anyway no?